config/routes.rb

```diff
@@ -20,9 +20,13 @@
     resource :vote, controller: 'comments/votes', only: [:create, :destroy]
   end
 
-  get ':username' => 'users/lists#index', as: :user_profile
+  get ':reference_id/:id/delete' => 'links#destroy', as: :destroy_link
+  get ':username/:list_id/:id/update' => 'references#edit_abstract', as: :edit_abstract
```

Comment:
so I swapped the link#destroy route with the resources rails method.  I was having trouble getting the delete verb to work properly, it kept mapping to the wrong route.  The resources method is probably better anyhow, especially if I add more actions.

For the edit_abstract, I kept it as a 'get' because it's just calling for the abstract form instead of updating.  I can switch the action to reference#edit, but I feel like that should be reserved for actually editing the whole reference instead of just the abstract.  

I read through http://guides.rubyonrails.org/routing.html but it didn't seem to give any advice about actions for editing a specific attribute.

This kind of gets at it but doesn't seem ideal: http://stackoverflow.com/questions/16952624/how-to-route-multiple-pages-to-edit-a-users-account-information-rails